tools/java/java-build/src/com/google/i18n/phonenumbers/BuildMetadataProtoFromXml.java

```diff
@@ -133,7 +140,7 @@ public boolean start() {
     }
 
     if (inputFile == null ||
-        outputDir == null ||
+        ((outputDir == null) == (outputFile == null)) ||
```

Comment:
Is the interior "==" supposed to be "||"? Or "&&"?

If it's indeed meant to be "==" (i.e., they are both null or both non-null), there should be a comment to that effect, or the logic rewritten to make that clear.